syntax = "proto3";

package payment;

service PaymentService {
    rpc CreatePayment(CreatePaymentRequest) returns (PaymentResponse);
    rpc ConfirmPayment(ConfirmPaymentRequest) returns (PaymentResponse);
}

message CreatePaymentRequest {
    int32 order_id = 1;
    string user_id = 2;
    double amount = 3;
    string currency = 4;
    string payment_method = 5;
    string provider = 6;
}

message ConfirmPaymentRequest {
    int32 payment_id = 1;
    string status = 2; // "PAID", "FAILED"
}

message PaymentResponse {
    int32 payment_id = 1;
    string external_id = 2;
    int32 order_id = 3;
    string status = 4;
}
